src/dns/hickory.rs

```diff
@@ -29,16 +33,37 @@ impl HickoryDnsResolver {
         Self {
             state: Default::default(),
             filter,
+            config: None,
         }
     }
+
+    pub fn with_config(mut self, config: ResolverConfig, opts: ResolverOpts) -> Self {
+        self.config = Some((config, opts));
+        self
+    }
+
+    fn new_resolver(&self) -> io::Result<TokioAsyncResolver> {
+        let (config, mut opts) = match self.config.clone() {
+            Some((config, opts)) => (config, opts),
+            None => system_conf::read_system_conf().map_err(|e| {
+                io::Error::new(
+                    io::ErrorKind::Other,
+                    format!("error reading DNS system conf: {e}"),
+                )
+            })?,
+        };
+
+        opts.cache_size = 500_000; // 500k entries
+        Ok(TokioAsyncResolver::tokio(config, opts))
+    }
```

Comment:
this might not be necessary... I could just pass the config through when we pass it, might try tidy this up